external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+if(${ENABLE_gauxc})
+    if(NOT "${gauxc_DIR}" STREQUAL "")
+        if("${IntegratorXX_DIR}" STREQUAL "")
+            set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+        endif()
+        set(ExchCXX_DIR ${gauxc_DIR}/../ExchCXX)
+    endif()
+
+    get_property(BLAS_LIBRARIES TARGET lapk PROPERTY INTERFACE_LINK_LIBRARIES)
+
+    find_package(gauxc CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== rediscover Libint2 to get its properties (specifically AM) ==#
+        # probably hacky; I'm guessing there's a more elegant sol'n
+        find_package(
+            Libint2
+            2.8.1
+            CONFIG
+        )
+
+        #== check that AMs of GauXC and Libint2 align ==#        
+        get_property(_am TARGET gauxc::gauxc PROPERTY GAUXC_CPU_SNLINK_MAX_AM)
```

Comment:
I suppose I should ask, in light of recent discussion surrounding runtime querying of GauXC AM capabilities in other comment chains. It sounds like we are making some progress towards that, so do we really need a compile-time cross-verification of Libint2 vs. GauXC supported AMs here? I feel like the cross-checking can be moved to run-time, as well, if so desired.

Reply:
In light of @susilehtola and @loriab 's comment viz hot swapability (L2 not, but I think GauXC can be made to be if it's not already) - I'd vote that this gets moved to runtime checks. The PR adding the runtime checker has been added.

Reply:
Excellent! I will do that, then.